### ISSUE TITLE
[Help Wanted] The zip downloaded using ServiceWorker is empty

### ISSUE THREAD

**Issue by TangBunao** [Touffy/client-zip#76]
Hello,
Thanks for the amazing library!

I am currently using your awesome library to download multiple files from URLs to a zip file. After reviewing the demo and related instructions you provided, I believe using Service Worker is more suitable for my needs. 

I built an HTTPS site using IIS on my machine and added your sample code, but it doesn't seem to work. 
When I clicked the download button, the browser only downloaded an empty zip file. When I checked at the network tab, I noticed that the browser was still downloading files, so I'm sorry to bother you and see if you could provide me with some help, such as the possible reasons and possible solutions to this issue.

My code is no different from the sample code (demo/worker. html), but I downloaded the client-zip/worker.js and dl-stream/worker.js to my local machine.
![image](https://github.com/Touffy/client-zip/assets/33479150/c86a620c-d968-4b13-ae88-a7f355f7875d)

Here is the GIF:
![client-zip-sample](https://github.com/Touffy/client-zip/assets/33479150/686139cd-0e08-4727-824f-34db5391854b)

I am not sure if I have described it clearly. If there is any ambiguity in my expression, please let me know.

Thank you very much for checking my question.

**Comment by Touffy** [comment]
It looks like the scripts are loaded and running, and there is no CSP blocking the download, since the network tab shows the fetch started by `dl-stream`. So your IIS server is not causing the issue.

Actually, I've tried it myself and the issue is definitely in my own code. Looking into it…

**Comment by Touffy** [comment]
Yup. It's a bug in `dl-stream`. Some of the Responses vanish, and when you give it a single URL, that Response is never in the iterator.

Until I fix the issue there, you can fall back to [the previous version of worker.js](https://github.com/Touffy/client-zip/blob/04af124f1e5012121acaa4a868b2b3ef64478c6c/demo/worker.js) — without dl-stream.

**Comment by Touffy** [comment]
I've fixed `dl-stream`. The way it was written, it would never yield the Responses for the last `length % 3` URLs. I didn't catch that before, because I used it to make archives with thousands of files and the exact number didn't matter.

Anyway… single-file (or any number of files) archives are now correctly generated by the current ServiceWorker demo without any other change. Just update your local copy of [`dl-stream`](https://github.com/Touffy/dl-stream).
